#! /bin/sh -e
## 05_config_list_member_options.dpatch by Tollef Fog Heen <tfheen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make it possible to set new_member_options as well. 
## DP: 

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argum
ent"
                exit 1;;
esac

exit 0
@DPATCH@

--- mailman-2.1.4.orig/bin/config_list
+++ mailman-2.1.4/bin/config_list
@@ -293,6 +293,11 @@
                                     in mm_cfg.OPTINFO.items()
                                     if validval & bitval]
                     gui._setValue(mlist, k, validval, fakedoc)
+                    # Ugly hack, but seems to be needed since
+                    # new_member_options isn't really a number in gui.
+                    # -- tfheen, 2003-12-06
+                    if k == "new_member_options":
+                        mlist.new_member_options = validval
             # BAW: when to do gui._postValidate()???
     finally:
         if savelist and not checkonly:
